src/main/host/syscall/handler/file.rs

```diff
@@ -104,9 +107,35 @@ impl SyscallHandler {
         Self::legacy_syscall(cshadow::syscallhandler_getdents64, ctx)
     }
 
-    #[log_syscall(/* rv */ std::ffi::c_int)]
-    pub fn lseek(ctx: &mut SyscallContext) -> SyscallResult {
-        Self::legacy_syscall(cshadow::syscallhandler_lseek, ctx)
+    #[log_syscall(/* rv */ std::ffi::c_int, /* fd */ std::ffi::c_uint,
+                  /* offset */ linux_api::posix_types::kernel_off_t, /* whence */ std::ffi::c_uint)]
+    pub fn lseek(
+        ctx: &mut SyscallContext,
+        fd: std::ffi::c_uint,
+        _offset: linux_api::posix_types::kernel_off_t,
+        _whence: std::ffi::c_uint,
+    ) -> Result<linux_api::posix_types::kernel_off_t, SyscallError> {
+        let desc_table = ctx.objs.thread.descriptor_table_borrow(ctx.objs.host);
+
+        let file = {
+            match Self::get_descriptor(&desc_table, fd)?.file() {
+                CompatFile::New(file) => file,
+                // if it's a legacy file, use the C syscall handler instead
+                CompatFile::Legacy(_) => {
+                    drop(desc_table);
+                    return Self::legacy_syscall(cshadow::syscallhandler_lseek, ctx)
+                        .map(Into::into);
+                }
+            }
+        };
+
+        match file.inner_file() {
+            File::Pipe(_) => Err(Errno::ESPIPE.into()),
+            _ => {
+                warn_once_then_debug!("lseek() is not implemented for non-pipes");
```

Comment:
This is slightly inaccurate since it's also implemented for legacy file descriptor types. Maybe just "not implemented for this type"?

Reply:
Updated to that string.